_posts/2022-10-06-audio-classification-torch/audio-classification.Rmd

```diff
@@ -0,0 +1,472 @@
+---
+title: "Audio classification with torch"
+description: >
+  Learn how to classify speech utterances with torch, making use of domain knowledge and deep learning. This post is a condensed version of the same-topic chapter in the book to be published with CRC Press, Deep Learning and Scientific Computing with R torch. 
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchaudio
+date: 2022-10-06
+categories:
+  - Torch
+  - R
+  - Audio Processing
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/squirrel.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE, fig.width = 6, fig.height = 6)
+```
+
+## Variations on a theme
+
+[Simple Audio Classification with Keras](https://blogs.rstudio.com/ai/posts/2018-06-06-simple-audio-classification-keras/), [Audio classification with Keras: Looking closer at the non-deep learning parts](https://blogs.rstudio.com/ai/posts/2019-02-07-audio-background/), [Simple audio classification with torch](https://blogs.rstudio.com/ai/posts/2021-02-04-simple-audio-classification-with-torch/): No, this is not the first post, on this blog, that introduces speech classification using deep learning. With two of those posts (the "applied" ones) it shares the general setup, the type of deep-learning architecture employed, and the dataset used. With the third, it has in common the interest in the ideas and concepts involved. Each of these posts has a different focus, -- should you read this one?
+
+Well, of course I can't say "no" here -- all the more that here, you have an abbreviated and condensed version of the same-topic chapter that is part of the book to be published by CRC Press, *Deep Learning and Scientific Computing with R `torch`*. By way of comparison with the previous post that used `torch`, written by the creator and maintainer of `torchaudio`, Athos Damiani, significant developments have taken place in the `torch` ecosystem, the end result being that the code got a lot easier (especially in the model training part). That said, let's end the preamble already, and plunge into the topic!
+
+## Inspecting the data
+
+We use the *speech commands* dataset (@abs-1804-03209) that comes with [`torchaudio`](https://github.com/curso-r/torchaudio). The dataset holds recordings of thirty different, one- or two-syllable words, uttered by different speakers; there are about 65,000 audio files overall. Our task will be to predict, from the audio solely, which of thirty possible words was pronounced.
```

Comment:
```suggestion
We use the *speech commands* dataset (@abs-1804-03209) that comes with [`torchaudio`](https://github.com/curso-r/torchaudio). The dataset holds recordings of thirty different one- or two-syllable words, uttered by different speakers. There are about 65,000 audio files overall. Our task will be to predict, from the audio solely, which of thirty possible words was pronounced.
```

Reply:
You know me, always out to destroy semi-colons 😈😉

Reply:
hehe :-)
Of course, I have mixed thoughts on this ;-)))
Mixed, because in the book, I've taken care to keep sentences short (or at least, that was the intention)